### ISSUE TITLE
Manually stopping a singleton service does not recover

### ISSUE THREAD

**Issue by dabooz** [open-horizon/anax#705]
Anax detects that the singleton is down, it tried to upgrade the service and then down grade it. Neither works because there is only 1 version of the service in the exchange, and then Anax gives up. Eventually governance wakes up and does this whole thing again. The singleton is never restarted.

**Comment by linggao** [comment]
booz: The SDR POC has 2 singleton services, one of them needed to be restarted because it was behaving incorrectly. The first thing I tried was to cancel the agreement, but that did not cause the singleton services to restart. I think we know this is the current behavior. Perhaps this is what we should fix. in any case, the next thing I tried was to manually stop the singleton. It has always been a design point that manually stopping a container will result in the agreement being cancelled and the containers to be restarted. In this case, that did not happen. The singleton was not restarted. The new agreement was created and the top level service was started. Anax did detect that the singleton container was down, then it tried to upgrade, then it tried to down grade, then it gave up. Eventually governance detected the container being down again and tried the upgrade/downgrade cycle again...over and over. It never tried to restart at the version we had before. This last bit is what I think is missing.
ling: For part 1 we left it that way by design.  But it is easy to fix if we decide to delete the service in singleton case. 
ling: I just checked the code, the container has code to restart a dependent service if detects the service is down.  And I did `docker stop` on my rpi to stop the sdr container, it was brought up automatically.  However, if the container is destroyed or failed to restart, it started the downgrading process as you have described.  We should recreate the container with the current version first in this case. The endless loop of downgrading should be fixed.
